### DISCUSSION TITLE
如何在获取用户信息后使用2份路由表中的其中一份？

### DISCUSSION THREAD

**Discussion by pangxzi** [umijs/umi#10288] [Q&A]
这种方法好像只能push，shift等方法，不能直接覆盖routes
![image](https://user-images.githubusercontent.com/52907799/212221999-bb8014dd-7e44-4845-9775-be4e18422978.png)

**Comment by fz6m** [comment]
一开始 0 路由，然后维护不同角色的不同路由表，全量 push 进去。

觉得 patch 太麻烦的话，用权限管控吧，access 插件、layout 里校验权限等思路，检测到没有权限就跳走。

**Reply by pangxzi** [reply]
好的，感谢
